### ISSUE TITLE
SPC i j adds a newline, and the cursor keeps it's previous state, but the minibuffer shows --INSERT--

### ISSUE THREAD

**Issue by duianto** [syl20bnr/spacemacs#6704]
## System
windows 10 version 1511 build 10586.494
emacs 24.5.1
spacemacs dev (downloaded 20160730), vim, standard, helm

## Issue:

When calling the command:
SPC i j
spacemacs/evil-insert-line-below

then one or more newlines are added below the current line 
(more if a COUNT argument is prepended).

(everything below also applies to this command:
SPC i k
spacemacs/evil-insert-line-above)

## Expected behavior:

The minibuffer shouldn't show --INSERT--,
because the cursor remains in it's previous state.

As stated in the functions comment:
https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Bdistributions/spacemacs-base/funcs.el#L97

> "Insert one of several lines below the current point's line without changing
> the current state and point position."

## Actual behavior:

The minibuffer shows: --INSERT--

## Similar commands where the issue doesn't occur

The --INSERT-- text doesn't appear in the minibuffer,
when these similar commands are called:

SPC i K
spacemacs/insert-line-above-no-indent
and
SPC i J
spacemacs/insert-line-below-no-indent

they are defined here:
https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Bdistributions/spacemacs-base/funcs.el#L535

## This seems to be what's causing the issue:

The command:
SPC i j
spacemacs/evil-insert-line-below
https://github.com/syl20bnr/spacemacs/blob/master/layers/%2Bdistribution/spacemacs-base/funcs.el#L93

saves the cursors position and state,
then it calls the command:
evil-open-below
https://bitbucket.org/lyro/evil/src/3f1515976813263540eb3cf3532f86a1df4cef92/evil-commands.el?at=default#evil-commands.el-2219

and it changes the cursor to the evil-insert-state,
and the minibuffer shows: --INSERT--.

When the evil-open-below command is done, 
then the cursors position and previous state are restored,
but the --INSERT-- text remains in the minibuffer.

### Other commands that aren't used:
When evil-open-below receives a COUNT argument,
then the insertion will be repeated COUNT times.

And the undo commands:
```
  (unless (eq evil-want-fine-undo t)
    (evil-start-undo-step))
  (push (point) buffer-undo-list)
```
probably has something to 
do with the repeated text insertion.

Therefore the evil-open-below command,
probably shouldn't be called from the:
SPC i j
spacemacs/evil-insert-line-below
command.

## Newline functions
Here are some newline functions, that might be useful 
when putting together a more relevant newline function 
for these commands:
SPC i j
spacemacs/evil-insert-line-below
and
SPC i k
spacemacs/evil-insert-line-above

funcs.el
https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Bdistributions/spacemacs-base/funcs.el#L535
line 535: spacemacs/insert-line-above-no-indent
line 549: spacemacs/insert-line-below-no-indent

evil-common.el
https://bitbucket.org/lyro/evil/src/3f1515976813263540eb3cf3532f86a1df4cef92/evil-common.el?at=default#evil-common.el-1892
line 1892: evil-insert-newline-above
line 1901: evil-insert-newline-below

simple.el
The github, emacs-mirror, simple.el file probably isn't the same as the version 
that came with emacs 24.5:
(extracted directory)\emacs-24.5-bin-i686-mingw32\share\emacs\24.5\lisp\simple.el
but here are the url / local line numbers to both versions.

newline
https://github.com/emacs-mirror/emacs/blob/master/lisp/simple.el#L389
local line 389

open-line
https://github.com/emacs-mirror/emacs/blob/master/lisp/simple.el#L467
local line 461

newline-and-indent
https://github.com/emacs-mirror/emacs/blob/master/lisp/simple.el#L625
local line 619

reindent-then-newline-and-indent
https://github.com/emacs-mirror/emacs/blob/master/lisp/simple.el#L636
local line 630

#### Another thing to consider is indentation:
The commands
SPC i j
spacemacs/evil-insert-line-below
and
SPC i k
spacemacs/evil-insert-line-above

probably should handle indentation, 
since their uppercase counterparts don't indent:

SPC i K
spacemacs/insert-line-above-no-indent
and
SPC i J
spacemacs/insert-line-below-no-indent

### Different ways of creating one or more newlines
These might not be relevant for this issue, 
but maybe there could be a transient state for creating newlines.

Restore the previous cursor position 
or 
move the cursor to the top or bottom of the created newlines,
depending initial direction.

Indent 
or
always place the cursor in the left most column.

Insert mode:
Change to insert mode at the top or bottom of the new line, based on initial direction,
or
when a COUNT argument is prepended, repeat the inserted text,
just like the evil-open-below (and ..-above) commands,
or
keep the current state.

New commands:
Create one or more newlines above and below the current line.

If more than one line is selected, add one or more newlines between each selected line.

**Comment by TheBB** [comment]
In the future it would help if you make your issues a bit more focused instead of reporting a specific problem and then turning it into a brainstorming session.

**Comment by TheBB** [comment]
https://github.com/syl20bnr/spacemacs/pull/6710

**Comment by duianto** [comment]
Sorry for the detour.

While writing the issue, i stumbled over a couple of things that could be corrected in funcs.el
misspellings, 
a missing comment, 
two possibly redundant comments, 
and 
two functions that could be swapped for logical and consistency reasons.

Should i report them as a single or multiple issues?

**Comment by TheBB** [comment]
Sounds like you could just PR the changes directly.

**Comment by duianto** [comment]
I have never done a pull request, but i watched a couple of tutorials, so i'll try it out.

**Comment by duianto** [comment]
I think i pressed the wrong button, i was going to cancel a comment, and i think it said, 
Close and comment.
Sorry about that.

**Comment by syl20bnr** [comment]
Thank you for taking time to write this issue, if you need help to create a PR you can check this document: https://github.com/syl20bnr/spacemacs/blob/develop/CONTRIBUTING.org
Don't hesitate to go on the gitter channel if you need more info.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
